runtime/lua/vim/lsp/client.lua

```diff
@@ -556,6 +560,10 @@ function Client:initialize()
       self.offset_encoding = self.server_capabilities.positionEncoding
     end
 
+    if result.serverInfo then
+      self.server_info = result.serverInfo
+    end
```

Comment:
Hmm I suppose the guard is actually redundant. We could also add some zero values instead of `nil` (e.g. `{name = '', version = ''}`) to make accessing it nicer/easier.

Reply:
> We could also add some zero values instead of `nil`

Doesn't that make testing for "not found" more difficult?